subprojects/plugins/src/main/java/org/gradle/api/plugins/WarPlugin.java

```diff
@@ -88,24 +106,34 @@ public void apply(final Project project) {
 
         PublishArtifact warArtifact = new LazyPublishArtifact(war, ((ProjectInternal) project).getFileResolver(), ((ProjectInternal) project).getTaskDependencyFactory());
         project.getExtensions().getByType(DefaultArtifactPublicationSet.class).addCandidate(warArtifact);
-        configureConfigurations(project.getConfigurations());
+        configureConfigurations((RoleBasedConfigurationContainerInternal) project.getConfigurations());
         configureComponent(project, warArtifact);
     }
 
-    public void configureConfigurations(ConfigurationContainer configurationContainer) {
```

Comment:
We can't remove public methods on non-internal types without a deprecation cycle

Reply:
Yes, but I think we should make an exception here.  This method not being `private` looks like a mistake - what's the use case for anyone other than us calling it?  There are already minor breaking changes in 8.0, this is just one more.

Reply:
> We can't remove public methods on non-internal types without a deprecation cycle

Reply:
Targeting 9.0 now, so this is a moot point.